### ISSUE TITLE
Implement closeaction=restart on ipsec.conf as an option during Phase 1 configuration

### ISSUE THREAD

**Issue by nwildner** [opnsense/core#3707]
Hi.

Some of the OPNSense updates require reboots(and that is mandatory).

This can be really problematic if you have a site-to-site IPSec environment using Mutual RSA and restart the server side(main office). Remote office will not come back online since there was a `close CHILD_SA` sent from the main office, issued by the ipsec service being stopped.

This will make passthrough the only connection available at the remote office(until ipsec is restarted), and this could be easily managed by adding the `closeaction=restart` to the `ipsec.conf` file at the remote office side.

I've made some tests by editing manually `ipsec.conf` and these are the results:

**Without `closeaction=restart` at the remote office - 0 connections up, 0 connecting.**

```
root@fw01xxx:/usr/local/etc # ipsec status
no files found matching '/usr/local/etc/strongswan.opnsense.d/*.conf'
Shunted Connections:
        pass:  10.1.140.0/24 10.0.1.0/24 192.168.1.0/24 === 10.1.140.0/24 10.0.1.0/24 192.168.1.0/24 PASS
Security Associations (0 up, 0 connecting):
  none

```
**With `closeaction=restart` manually configured at `/usr/local/etc/ipsec.conf`**. Ipsec status shows that it's trying to connect while main office is rebooting (1 connecting).

```
root@fw01xxx:/usr/local/etc # ipsec status
no files found matching '/usr/local/etc/strongswan.opnsense.d/*.conf'
Shunted Connections:
        pass:  10.1.140.0/24 10.0.1.0/24 192.168.1.0/24 === 10.1.140.0/24 10.0.1.0/24 192.168.1.0/24 PASS
Security Associations (0 up, 1 connecting):
        con1[2]: CONNECTING, 10.0.1.10[%any]...138.xx.xx.xx[%any]
```

After 2 or 3 minutes: 

```
root@fw01xx:/usr/local/etc # ipsec status
no files found matching '/usr/local/etc/strongswan.opnsense.d/*.conf'
Shunted Connections:
        pass:  10.1.140.0/24 10.0.1.0/24 192.168.1.0/24 === 10.1.140.0/24 10.0.1.0/24 192.168.1.0/24 PASS
Security Associations (1 up, 0 connecting):
        con1[2]: ESTABLISHED 25 seconds ago, 10.0.1.10[fw01yyy@yy.yy.yy]...138.xx.xx.xx[site2site@yy.yy.yy]
        con1{2}:  INSTALLED, TUNNEL, reqid 2, ESP in UDP SPIs: c1eab391_i cc38aa4d_o
        con1{2}:   10.1.140.0/24 === 0.0.0.0/0
```

Additional information at the [Strongswan Docs](https://wiki.strongswan.org/projects/strongswan/wiki/ConnSection):

> closeaction = none | clear | hold | restart
> 
> defines the action to take if the remote peer unexpectedly closes a CHILD_SA (see dpdaction for
> meaning of values). A closeaction should not be used if the peer uses reauthentication or uniqueids checking, as these events might trigger the defined action when not desired. 
> Prior to 5.1.0, closeaction was not supported for IKEv1 connections.

---------
**So, here comes my request:** If possible, create an option for Phase 1 IPSec, with something like "Restart this Connection when receiving close CHILD_SA from remote" and at the help, explain that this needs to be enabled only at the remote offices, not at the side where IPSec is configured with "Respond Only" connection method. Checking this option would change the default `none` to `restart` to `closeaction=`

**Comment by AdSchellevis** [comment]
We merged https://github.com/opnsense/core/pull/3851 some time ago, which adds support for manual configuration of `dpdaction`.  Although it's not the same as `closeaction` for quite some scenarios this also solves the issue of not being reconnected automatically.

**Comment by nwildner** [comment]
> We merged #3851 some time ago, which adds support for manual configuration of `dpdaction`. Although it's not the same as `closeaction` for quite some scenarios this also solves the issue of not being reconnected automatically.

Yup. The real issue would be if the main office restart due to version upgrades, sending close CHILD_SA to all peers, and dpdaction is basically the oposite, when branch offices get a timeout, what action to take...

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/core/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.

**Comment by nwildner** [comment]
> 
> 
> We merged #3851 some time ago, which adds support for manual configuration of `dpdaction`. Although it's not the same as `closeaction` for quite some scenarios this also solves the issue of not being reconnected automatically.

Hey @AdSchellevis , just to let you know: we've migrated our core sdwan to another [proprietary technology](https://forum.opnsense.org/index.php?topic=13543.msg88551#msg88551). I think this vendor change sumed up with  #3851 dpd action handling solved our issues.

I can confirm that disabling all Phase 2 tunels at our main office(non-OPNSense), creates a `close CHILD_SA` event(logs below), but, as soon as we reenable this tunel, OPNSense dpd handles the Phase 2 reconection as expected and implemented on. No need to reopen this issue. This reply of mine is informative only.

```
2020-09-24T08:59:08 | charon[91760] | 12[ENC] <con1\|3> generating INFORMATIONAL response 1 [ D ]
2020-09-24T08:59:08 | charon[91760] | 12[IKE] <con1\|3> CHILD_SA closed
2020-09-24T08:59:08 | charon[91760] | 12[IKE] <con1\|3> sending DELETE for ESP CHILD_SA with SPI c5ce0ea5
2020-09-24T08:59:08 | charon[91760] | 12[IKE]  <con1\|3> closing CHILD_SA con1{9} with SPIs c5ce0ea5_i (1396  bytes) b1dc7ec1_o (3984 bytes) and TS 10.254.0.0/16 === 0.0.0.0/0
```
